### ISSUE TITLE
Crashes on -[CTInboxController unreadMessages] CTInboxController.m, line 163

### ISSUE THREAD

**Issue by atom-wintermute** [CleverTap/clevertap-ios-sdk#163]
**Describe the bug**
- dozens of crashes in Firebase Crashlytics

**To Reproduce**
- where is no straight way to reproduce it

**Expected behaviour**
- SDK doesn't crash

**Environment (please complete the following information):**
 - Xcode 13.1.0
 - CleverTap SDK 3.10.0
 - Device: [e.g. iPhone12, iPhone7]
 - OS: [e.g. iOS 13.3]

**Possible Solution**
- all accessors to database should be wrapped with `privateContext`

**Additional context**

```
MAIN THREAD - CRASHED

libobjc.A.dylib
objc_msgSend
CoreFoundation
CFBasicHashAddValue
CoreFoundation
__NSOrderedSetM_new
CoreData
-[_NSFaultingMutableOrderedSet willReadWithContents:]
CoreData
-[_NSFaultingMutableOrderedSet count]
Foundation
-[NSOrderedSet(NSPredicateSupport) filteredOrderedSetUsingPredicate:]
CleverTapSDK
-[CTInboxController unreadMessages] CTInboxController.m:163
CleverTapSDK
-[CleverTap getUnreadInboxMessages] CleverTap.m:4186
```

**Comment by akashvercetti** [comment]
@litongde we have implemented some possible fixes for the inbox issue and would be releasing them along with iOS 16 support tomorrow. 👍

**Comment by akashvercetti** [comment]
@atom-wintermute we have released a new version v4.1.2 with possible inbox fixes. Please update to the latest version and see if it works out.
